plugins/ws.js

```diff
@@ -3,6 +3,28 @@ var URL = require('url')
 var pull = require('pull-stream/pull')
 var Map = require('pull-stream/throughs/map')
 
+function safe_origin (origin, address, port) {
+
+  //if the connection is not localhost, we shouldn't trust
+  //the origin header. So, use address instead of origin
+  //if origin not set, then it's definitely not a browser.
+  if(!(address === '::1' || address === '128.0.0.1') || origin == undefined)
```

Comment:
128? I guess it should be `127.0.0.1` right? Is this a resolved address? will `localhost` work as well?